### PR TITLE
Make TTL optional (again)

### PR DESCRIPTION
**Pull request by rebstar6**
Changes in this pull request:
- make batch ttl optional
- if not provided, add should go to default, update should remain unchanged

### CODE REVIEW

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py [hunk 1]
```diff
@@ -502,6 +502,64 @@ def test_create_batch_change_without_owner_group_id_succeeds(shared_zone_test_co
     finally:
         clear_zoneid_rsid_tuple_list(to_delete, client)
 
+
+def test_create_batch_change_with_missing_ttl_returns_default_or_existing(shared_zone_test_context):
+    """
+    Test creating a batch change without a ttl fails
```
Comment (britneywright) [inline modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: should this be "succeeds" rather than "fails"?

**Comment by britneywright** [comment]
looks good to me. Just need to update the ttl description in the docs https://www.vinyldns.io/api/batchchange-model.html#singleaddchange-attributes

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py (hunk 1 above)
Comment (rebstar6) [reply modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: oh lol yeah

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py [hunk 2]
```diff
@@ -502,6 +502,66 @@ def test_create_batch_change_without_owner_group_id_succeeds(shared_zone_test_co
...
+
+        result = client.create_batch_change(batch_change_input, status=202)
+        completed_batch = client.wait_until_batch_change_completed(result)
+        record_set_list = [(change['zoneId'], change['recordSetId']) for change in completed_batch['changes']]
+        to_delete = set(record_set_list)
+
+        updated_record = client.get_recordset(record_set_list[0][0], record_set_list[0][1])['recordSet']
+        assert_that(updated_record['ttl'], is_(300))
+
+        # TODO: this assertion will fail if the default TTL is changed
```
Comment (rebstar6) [inline modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: dont love this...

**Pull request merged**